### PR TITLE
[FIX] stock_account: correct return value for avco real_time valuation

### PR DESCRIPTION
**Pull request by robinengels**
Current behavior:
If you use AVCO and Automatic stock valuation in anglo-saxon
accounting. When you make 2 purchase order for the same product with
different unit price and make a return for one of the 2 PO the return
value should be the same as the PO and not the average of the 2 PO.

Steps to reproduce:
- Make sure you are using anglo-saxon accounting
- Create a Product A with product category A that use AVCO and
  automatic valuation.
- Create a PO for product A with a unit price of 10€, validate
  the PO and receipt the product
- Create a PO for product A with a unit price of 20€, validate
  the PO and receipt the product
- Create a return for the second PO, validate it and click on
  the valuation smart button
- The value should be 20€ but it's 15€ (the average)

opw-2961395
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by agr-odoo** [comment]
Well, according to the [documentation](https://www.odoo.com/documentation/15.0/applications/finance/accounting/others/inventory/avg_price_valuation.html#further-thoughts-on-anglo-saxon-mode), upon returning an item, the system shall perform these actions:
1. Hit the stock input account with the initial price
2. Hit the price difference account with the price difference of 1. if any
As far as I can tell, Avg Cost and Inventory value are not affected by the outgoing operations, even if it is a return.

I've reproduced the documentation example and on the accounting side it seems working as expected, provided that the return RBILL is issued. as only then, the price difference account is hit and the stock input is balanced (see screenshot below). 
screenshot_024

Reproducing again with the commit use case
screenshot_025
Completing the flow will leave a product valuation of 15, which, according to the documentation is correct, as it seems that only incoming transfer should impact the valuation.

The following is what happens when applying this change
screenshot_027
Everything is balanced as you are reducing the stock valuation but the price difference account is not hit and it is quite different from before.

In conclusion the documentation looks a bit misleading as the accounting operations are not performed on return, but when posting the RBILL.
As I'm not sure what the Inventory Valuation (AVG cost of the product) should be after return it may be worth to ask the opinion of @amoyaux

**Comment by robinengels** [comment]
@amoyaux

**Comment by robinengels** [comment]
@amoyaux

**Comment by amoyaux** [comment]
The right place to fix the price of the layer should be in get_price_unit

**Comment by robinengels** [comment]
@amoyaux The `get_unit_price` function is not used when creating out_svl, should I try to use it ?

**Comment by robinengels** [comment]
@amoyaux up

**Comment by robinengels** [comment]
@amoyaux up

**Comment by robinengels** [comment]
@amoyaux ping

**Comment by robinengels** [comment]
Need to add a price difference account on the product category as explained in the doc here (https://www.odoo.com/documentation/15.0/applications/finance/accounting/others/inventory/avg_price_valuation.html#further-thoughts-on-anglo-saxon-mode)
